### PR TITLE
Sliding windows return their start and ends.

### PR DESCRIPTION
**Pull request by Ward9250**
This modifies the code for sliding windows slightly, such that in a for loop now you get a tuple of values. The first value is the start and end position of the window, and the second value is the view of the sequence / array itself in that region.

### CODE REVIEW

File: src/util/windows.jl [hunk 1]
```diff
@@ -87,8 +87,9 @@ end
 
 @inline function Base.next(it::EachWindowIterator, state::Integer)
     i = state
-    window = sub(it.data, i:i + it.width - 1)
-    return window, i + it.step
+    region = i:i + it.width - 1
```
Comment (bicycle1885) [inline src/util/windows.jl]: If the window size is fixed, returning the end position is not needed, which is also compatible with our each-kmer iterator.

Comment (Ward9250) [inline src/util/windows.jl]: I think this is what you meant: I altered this so now it returns (i, window).

Comment (bicycle1885) [inline src/util/windows.jl]: Exactly.

File: test/util/windows.jl [hunk 2]
```diff
@@ -104,19 +104,19 @@ end
                 @test missed(ribitr) == expectedMissed
                 @test missed(aasitr) == expectedMissed
                 for win in arrres
-                    @test length(win) == winwidth
+                    @test length(win[2]) == winwidth
```
Comment (bicycle1885) [inline test/util/windows.jl]: Testing `win[1]` is desirable.

Comment (Ward9250) [inline test/util/windows.jl]: Added tests for `win[1]`, will see if it passes suite.

**Pull request merged**